### PR TITLE
fix: remove padding from state used in OIDC authorise

### PR DESCRIPTION
**Pull request by nsarupr**
## Description

> Cognito returns **Malformed URI** for redirect when User tries to Login. Due to this, browser ends up throwing `400 Bad Request`. In order to fix this, we have removed the padding from the `state` query parameter which we send to the OIDC, in order to avoid the cases where the additional `=` will lead to malformed URIs being created. Read [here](https://stackoverflow.com/questions/6916805/why-does-a-base64-encoded-string-have-an-sign-at-the-end) for Additional information on the Base64 encoding and padding.
> Also, we are changing the delimiter from `=` to `-` when server tries to find the redirect URI for other use cases.
> Server uses `,` in order to split the state to get the `origin value`. Now we will use `@` instead of `,`.

> TL;DR, remove `=` and `,` in order to avoid malformed URI strings.

Fixes https://github.com/appsmithorg/appsmith/issues/19692

Media
> A video or a GIF is preferred. when using Loom, don’t embed because it looks like it’s a GIF. instead, just link to the video

## Type of change

- Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

**Comment by trishaanand** [comment]
Tests on EE with the same changes have passed (except an unrelated failure for upgrade) : https://github.com/appsmithorg/appsmith-ee/actions/runs/4113772980
